### PR TITLE
docs: Add stub articles for search guides

### PR DESCRIPTION
**Pull request by m-allanson**
## Description

Update the existing search guide to act as an overview for more specific guides.

I've also updated the [[guidelist]] handler to remove the hardcoded paths (cc @jlengstorf)

screenshot 2019-01-02 at 14 25 47

## Related Issues

Refs #10113

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@m-allanson i'm working on documentating how to search with reactivesearch. I can expand the document i'm producing for both of elasticunr and js-search also. Algolia for me it's not something i really not that confortable, so i'll refrain from expanding that stub and probably fumble fingering it.

File: www/src/templates/template-docs-markdown.js [hunk 1]
```diff
@@ -15,9 +15,26 @@ import docsHierarchy from "../data/sidebars/doc-links.yaml"
 // Find the guides in the sidebar YAML.
 const guides = docsHierarchy.find(group => group.title === `Guides`).items
 
+// Search through the guides and their children to find the first item that
+// matches the given slug
+const findBySlug = (guides, slug) => {
```
Comment (DSchau) [inline www/src/templates/template-docs-markdown.js]: This looks better, but I wonder if we could make it less side-effecty with a `.find` and perhaps a flatten? Here's a start:

```js
const flatten = arr => arr.reduce((merged, item) => merged.concat(Array.isArray(item.items) ? flatten(item.items) : item), [])

const findBySlug = (guides, slug) => {
  return flatten(guides)
    .find(item => item.link === slug)
}
```

Comment (DSchau) [reply www/src/templates/template-docs-markdown.js]: Presuming we have a structure like (for guides):

```js
const guides = [
    {
      link: '/test'
    },
    {
      items: [
        {
          link: '/test2'
        }
      ]
    }
  ]
```

this seems to work just fine!

File: www/src/data/sidebars/doc-links.yaml [hunk 2]
```diff
@@ -191,6 +191,13 @@
       items:
         - title: Adding search
           link: /docs/adding-search/
+          items:
+            - title: "Adding search with Algolia*"
```
Comment (LekoArts) [inline www/src/data/sidebars/doc-links.yaml]: The apostrophes `"` aren't used anywhere else, so you should probably don't use them for coherence

Comment (jlengstorf) [reply www/src/data/sidebars/doc-links.yaml]:
```suggestion
            - title: Adding search with Algolia*
```

File: www/src/data/sidebars/doc-links.yaml [hunk 3]
```diff
@@ -191,6 +191,13 @@
...
+              link: /docs/adding-search-with-algolia/
+            - title: "Adding search with elasticlunr*"
```
Comment (jlengstorf) [inline www/src/data/sidebars/doc-links.yaml]:
```suggestion
            - title: Adding search with elasticlunr*
```

File: www/src/data/sidebars/doc-links.yaml [hunk 4]
```diff
@@ -191,6 +191,13 @@
...
+              link: /docs/adding-search-with-elasticlunr/
+            - title: "Adding search with js-search*"
```
Comment (jlengstorf) [inline www/src/data/sidebars/doc-links.yaml]:
```suggestion
            - title: Adding search with js-search*
```

File: www/src/data/sidebars/doc-links.yaml (hunk 2 above)
Comment (m-allanson) [reply www/src/data/sidebars/doc-links.yaml]: Thanks, I could have sworn I copied the quote style from elsewhere in the document, but clearly I was just making things up 🤔

File: www/src/templates/template-docs-markdown.js (hunk 1 above)
Comment (m-allanson) [reply www/src/templates/template-docs-markdown.js]: I had a little play around with this but I don't think flatten is right here, as we want to grab the child items from the item with the matching slug. I rejigged some other bits while playing around with this though, thank you 👍

**Comment by jonniebigodes** [comment]
ok, so with this merge now my pull request #11030 is basically throwing merge conflicts with what is now the master. If you don't mind, a little feedback would be appreciated on this matter

**Comment by m-allanson** [comment]
Sorry I missed your comment earlier, I've fixed up the merge conflicts on your branch.

**Pull request merged**